### PR TITLE
Support API like 'json.dump' and 'json.dumps'

### PR DESCRIPTION
**Pull request by rruc**

### CODE REVIEW

**Comment by mitsuse** [comment]
@rruc Thanks for your pull-request! Please wait for a while.

File: typedjson/dumping.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+from __future__ import annotations
+
+from typing import Any
+from typing import Dict
+from typing import IO
+from typing import TypeVar
+
+Decoded = TypeVar('Decoded')
```
[the remark this answers is not in the document]
Comment (mitsuse) [reply typedjson/dumping.py]: As a future work, we can provide a decorator or metaclass to conform `typedjson.Encodable` automatically.

**Comment by rruc** [comment]
Thanks for the review. I'll make a new pull request in a while.

**Comment by mitsuse** [comment]
@rruc I have just backported for Python 3.6.x, so Travis CI runs tests on both Python 3.7.x and 3.6.x. Please rebase your branch with the master branch.

**Comment by rruc** [comment]
> @rruc I have just arranged your commits with rebasing and cherry-picking:
> 
> https://github.com/mitsuse/typedjson-python/compare/pq/rucc/support-dump-and-dumps
> 
> I'm planning to merge these changes after that I added two modifications below:
> 
>     * Can I write your name in [CHANGELOG.md]?

Yes

> 
>     * I will add `typedjson.encodable` decorator to use `typedjson.Encodable` instead of `Any`.

Ok!

> 
> 
> The rest parts of commits will be merged into the master branch later because they mainly consist of cosmetic changes.

Ok!

**Comment by mitsuse** [comment]
@rruc  I decided to postpone adding`@typedjson.encodable` decorator because it seems to require mypy-plugin. The changes of this pull-request has been merged into the master as `pq/rucc/support-dump-and-dumps` branch. Thanks for your contributions!

**Pull request closed without merging**